src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -370,116 +479,138 @@ private static Details getDetails(DetailAST ast) {
                 if (lcurly.getType() == TokenTypes.SLIST) {
                     rcurly = lcurly.getLastChild();
                 }
-                break;
-            case TokenTypes.LITERAL_ELSE:
-            case TokenTypes.LITERAL_FINALLY:
+
+            }
+            else {
                 shouldCheckLastRcurly = true;
                 nextToken = getNextToken(ast);
                 lcurly = ast.getFirstChild();
                 if (lcurly.getType() == TokenTypes.SLIST) {
                     rcurly = lcurly.getLastChild();
                 }
-                break;
-            case TokenTypes.CLASS_DEF:
+            }
+            final Details details = new Details();
+            details.rcurly = rcurly;
+            details.lcurly = lcurly;
+            details.nextToken = nextToken;
+            details.shouldCheckLastRcurly = shouldCheckLastRcurly;
+            return details;
+        }
+
+        /**
+         * Collects validation details for CLASS_DEF, METHOD DEF, CTOR_DEF, STATIC_INIT, and
+         * INSTANCE_INIT.
+         * @param ast a {@code DetailAST} value
+         * @return an object containing all details to make a validation
+         */
+        private static Details getDetailsForDefs(DetailAST ast) {
+            DetailAST rcurly = null;
+            final DetailAST lcurly;
+            final DetailAST nextToken;
+            final int tokenType = ast.getType();
+            if (tokenType == TokenTypes.CLASS_DEF) {
```

Comment:
Yeah, but wouldn't the need for switch also depend on how the details get collected. Perhaps we might not need many branches for different token ? I feel this can be managed in issue #3547 ? What do you think ?